Deprecate the Typeclasses Filtered Unification flag. (#15752)
coq/coq: doc/changelog/07-vernac-commands-and-options/15752-deprecate-filtered-tc.rst

- **Deprecated:**

>   the :flag:`Typeclasses Filtered Unification` flag. Due to
>   its alleged unusability, it is unlikely to be used in the wild

Reviewer: ```suggestion
  the :flag:`Typeclasses Filtered Unification` flag. Due to
  a buggy implementation, it's unlikely this is being used
```
